src/objects/abstract/dsm.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2019 CyPhyHouse. All Rights Reserved.
+
+from abc import ABC, abstractmethod
+
+
+class dsm(ABC):
+    """
+    abstract distributed shared memory object
+    __name : variable name
+    __data_type : data type
+    """
+
+    def __init__(self, name: str, data_type: type):
+        """
+
+        :param name: variable name
+        :type name: str
+
+        :param data_type: variable datatype
+        :type data_type: type
+
+        """
+        self.__name = name
+        self.__data_type = data_type
+
+    def __repr__(self):
+        """
+        string representation
+        """
+        return str(self.name)
+
+        # ------------ MEMBER ACCESS METHODS --------------
+
+    @property
+    def name(self) -> str:
+        return self.__name
+
+    @name.setter
```

Comment:
Why are setter methods for `__name` and `data_type` necessary?

Reply:
not necessary, can delete.